File: JavaProject/Manager.java
import java.util.Date;
import java.util.List;
public class Manager extends Employee implements I_Receptionist, I_Manager, I_Repairman, I_Cleaner {
	
	private Client client;
	private List<Room> rooms;
	public void Make_reservation(String client_id, Date start_date, Date end_date, int room_number){
		
	}
	public void Take_payment(String client_id, double value, boolean card){
		
	}
	public void Register_client(String last_name, String first_name, String client_id, String email){
		
	}
	public List<Room> Find_free_room(Date start_date, Date end_date, int capacity){
		
		return rooms;
	}
	public void Check_in_client(String reservation_id){
		
	}
	public void Cancel_reservation(String reservation_id){
		
	}
	public void Check_out_client(String reservation_id){
		
	}
	public Client Find_client(String client_id){

		return client;
	}
	public void Call_menager(String place){
		
	}
	public void report_need_for_cleaning(int room_number){
		
	}
	public void report_need_for_repair(int room_number){
		
	}
	public void report_need_for_security_intervention(int room_number){
		
	}
	public void Report_execution_of_cleaning(int room_number){
		
	}
	public void Report_execution_of_repair(int room_number){
		
	}
	public void View_reports(){
		
	}
	public void Print_and_archive_report(Date date){
		
	}
	public void Add_daily_report(){
		
	}
	public void PayOut_salary(String employee_ID){
		
	}
	public void Give_bonus(String employee_id, double bonus){
		
	}
	public void Calculate_salary(String employee_id){
		
	}
	public void Fire_employee(String employee_id){
		
	}
	public void Check_hotel_statistics(){
		
	}
	public void Call_employee(String employee_id){
		
	}
	public void Warn_employee(String employee_id){
		
	}


	
}

File: CSharpProject/Hotel/Model/Manager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HotelProject.Model
{
    class Manager : Employee, IManager, IReceptionist, IRepairman, ICleaner
    {
        public Manager() : base() { }

        public Manager(String firstName, String lastName, String id, String email, String password) : base(firstName, lastName, id, email, password) { }

        public void AddDailyReport()
        {
            throw new NotImplementedException();
        }

        public void CalculateSalary(string employeeID)
        {
            throw new NotImplementedException();
        }

        public void CallEmployee(string employeeID)
        {
            throw new NotImplementedException();
        }

        public void CallManager(string place)
        {
            throw new NotImplementedException();
        }

        public void CancelReservation(string reservationID)
        {
            throw new NotImplementedException();
        }

        public void CheckInClient(string reservationID)
        {
            throw new NotImplementedException();
        }

        public void CheckOutClient(string reservationID)
        {
            throw new NotImplementedException();
        }

        public Client FindClient(string clientID)
        {
            throw new NotImplementedException();
        }

        public Room[] FindFreeRoom(DateTime startDate, DateTime endDate, int capacity)
        {
            throw new NotImplementedException();
        }

        public void FireEmployee(string employeeID)
        {
            throw new NotImplementedException();
        }

        public void GiveBonus(string employeeID, double bonus)
        {
            throw new NotImplementedException();
        }

        public void MakeReservation(string clientID, DateTime startDate, DateTime endDate, int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void PayOutSalary(string employeeID)
        {
            throw new NotImplementedException();
        }

        public void PrintAndArchiveReport(DateTime date)
        {
            throw new NotImplementedException();
        }

        public void RegisterClient(string lastName, string firstName, string clientID, string email)
        {
            throw new NotImplementedException();
        }

        public void ReportExecutionOfCleaning(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void ReportExecutionOfRepair(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void ReportNeedForCleaning(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void ReportNeedForRepair(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void ReportNeedForSecurityIntervention(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void TakePayment(string clientID, double value, bool card)
        {
            throw new NotImplementedException();
        }

        public void ViewReports()
        {
            throw new NotImplementedException();
        }

        public void WarnEmployee(string employeeID)
        {
            throw new NotImplementedException();
        }
    }
}
